providers/src/airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -757,6 +759,16 @@ async def _load_config(self):
             await async_config.load_kube_config_from_dict(self.config_dict)
             return async_client.ApiClient()
 
+        if kubeconfig_path is not None:
+            self.log.debug("loading kube_config from: %s", kubeconfig_path)
+            self._is_in_cluster = False
+            await async_config.load_kube_config(
+                config_file=kubeconfig_path,
+                client_configuration=self.client_configuration,
+                context=cluster_context,
+            )
+            return async_client.ApiClient()
+
```

Comment:
This could be merged with the block below, like if kubeconfig_path is provided then load the file, if not, and kubeconfig is provided, write it to temp file and use it as  kubeconfig_path. (you may need to move the logic to an inner method to call it inside the temp file context and outside it)

Reply:
thanks for the review :+1: 

my fix is a reflect of the current implementation of the not async version , I had in mind to only fix the issue , not refactor the KPO code duplication , is that okay ?

Reply:
wdyt @hussein-awala ?

Reply:
I will do this in a separate PR